### PR TITLE
feat: add SetIf APIs

### PR DESCRIPTION
**Pull request by pgautier404**
This commit adds `SetIf` APIs to the SDK. Note that all of the `.go` files in `internal/protos` are generated code and do not require review.

### CODE REVIEW

**Review by pratik151192 — DISMISSED** [review]
looks great!

File: momento/cache_client.go [hunk 1]
```diff
@@ -32,8 +32,20 @@ type CacheClient interface {
 	Increment(ctx context.Context, r *IncrementRequest) (responses.IncrementResponse, error)
 	// Set sets the value in cache with a given time to live (TTL)
 	Set(ctx context.Context, r *SetRequest) (responses.SetResponse, error)
-	// Set sets the value in cache with a given time to live (TTL) if not already present
+	// SetIfNotExists sets the value in cache with a given time to live (TTL) if key is not already present
 	SetIfNotExists(ctx context.Context, r *SetIfNotExistsRequest) (responses.SetIfNotExistsResponse, error)
+	// SetIfAbsent sets the value in cache with a given time to live (TTL) if key is not already present
+	SetIfAbsent(ctx context.Context, r *SetIfAbsentRequest) (responses.SetIfAbsentResponse, error)
```
Comment (anitarua) [inline momento/cache_client.go:38]: SetIfNotExists and SetIfAbsent are meant to do the same thing, right?

**Review by anitarua — APPROVED** [review]
lgtm!

File: momento/cache_client.go (hunk 1 above)
Comment (pgautier404) [reply momento/cache_client.go:38]: Yes, I believe we're going to deprecate SetIfNotExists.

Comment (cprice404) [reply momento/cache_client.go:38]: this actually raises a good question. we should discuss this one in slack.

**Review by anitarua — DISMISSED** [review]
🛳️

**Review by pratik151192 — DISMISSED** [review]

**Review by anitarua — APPROVED** [review]

**Pull request merged**